### PR TITLE
samples: cellular: modem_shell: Add support for counting pulses

### PR DESCRIPTION
**Pull request by tokangas**
Added command `gpio_count` to count pulses on a GPIO pin. When pulse counting has been enabled, LED 2 on nRF91 Series DKs also shows the state of the pin input.

### CODE REVIEW

File: samples/cellular/modem_shell/README.rst
Comment (peknis) [inline samples/cellular/modem_shell/README.rst:828]:
```suggestion
When pulse counting has been enabled, **LED 2** on nRF91 Series DKs shows the state of the pin input.
```

**Review by peknis — APPROVED** [review]
Approved with a minor nit.

File: samples/cellular/modem_shell/src/gpio_count/gpio_count.h
Comment (trantanen) [inline samples/cellular/modem_shell/src/gpio_count/gpio_count.h:15]: I probably would have left all these functions into `gpio_count_shell.c` as we have done in the small command modules. But I'll leave it to you to decide.

File: samples/cellular/modem_shell/README.rst
Comment (trantanen) [inline samples/cellular/modem_shell/README.rst:828]: It seems there is a limitation that a single GPIO can be counted at a time. This should be mentioned somewhere.

File: samples/cellular/modem_shell/README.rst
Comment (trantanen) [inline samples/cellular/modem_shell/README.rst:827]: We should probably add to the beginning and the first sentence modified to fit better:

MoSh command: ``gpio_count``

File: samples/cellular/modem_shell/README.rst
Comment (trantanen) [inline samples/cellular/modem_shell/README.rst:841]: Maybe add `gpio_count disable` command and get after that if it returns a value (I think it does based on code).

**Review by trantanen — CHANGES_REQUESTED** [review]

File: samples/cellular/modem_shell/src/gpio_count/gpio_count.h
Comment (tokangas) [reply samples/cellular/modem_shell/src/gpio_count/gpio_count.h:15]: I did consider whether to have separate file for the GPIO implementation or not. It is there now and moving the implementation to `gpio_count_shell.c` would cause some work and I don't see any clear reason for doing it. So, I'm keeping it as it is.

File: samples/cellular/modem_shell/README.rst
Comment (tokangas) [reply samples/cellular/modem_shell/README.rst:827]: I did not notice that most of the commands are documented like this. Not all are though, at least `sleep`, `uart` or `heap`. I will update all of these.

File: samples/cellular/modem_shell/README.rst
Comment (hiltunent) [inline samples/cellular/modem_shell/README.rst:835]: Please be more specific about how the counting actually happens. I am guessing you are counting rising edges?

**Review by trantanen — APPROVED** [review]

File: samples/cellular/modem_shell/README.rst
Comment (tokangas) [reply samples/cellular/modem_shell/README.rst:835]: Good point, I'll add it. Yes, it is counting rising edges.

**Review by hiltunent — APPROVED** [review]

File: samples/cellular/modem_shell/README.rst
Comment (peknis) [inline samples/cellular/modem_shell/README.rst:838]:
```suggestion
When pulse counting is enabled, **LED 2** on the nRF91 Series DKs shows the state of the pin input.
```

**Review by peknis — APPROVED** [review]
Approving with minor suggestion.

**Pull request merged**
